drivers/gpio/gpio_stm32.c

```diff
@@ -627,6 +627,9 @@ static int gpio_stm32_init(struct device *device)
 
 #define GPIO_DEVICE_INIT(__name, __suffix, __base_addr, __port, __cenr, __bus) \
 	static const struct gpio_stm32_config gpio_stm32_cfg_## __suffix = {   \
+		.common = {						       \
+			.available = (gpio_port_pins_t)-1,		       \
```

Comment:
I actually spent some time to understand how substracting 1 to gpio_port_pins_t could work. Anyway to disambiguate ?

Then, STM32 GPIO ports actually control 16 bits, so we should use `GPIO_AVAILABLE_PINS` macro, checking ngpios value. Bright side is that code will be ready for update to a range version.
Adding the whole fixup code is a bit cumbersome, so I'm preparing a commit to spare you the effort.

Last, I've tried to set `ngpios` by default to 16 `st,stm32-gpio.yaml`, but it is actually not possible to override the 32 value defined in `gpio-controller`. Is there a strong requirement to prevent this?
I'm not quite in favor of spreading dts values in bindings, but this is indeed handy for this kind of things.

Reply:
I suppose `(gpio_port_pins_t)(-1)` might provide a clue where necessary.

If the GPIO ports control 16 bits, then this should be `GPIO_AVAILABLE_PINS(16U)`.  I could make that change.  IIRC I went with the literal because the implementation depends on fixups to get the right devicetree value and I didn't want to add all of those (I ultimately had to for IMX because reasons, but it's not good).  I'd welcome a patch that turns it into what you want.

I don't see any reason not to allow overriding an existing default; the whole dts binding description architecture is pretty ad hoc.  This PR requires and includes #21271; if that doesn't get merged soon I'll need to push for it.  If you make a PR to support default override we can include it here too.

Reply:
+1 for `(gpio_port_pins_t)(-1)`
`GPIO_AVAILABLE_PINS(16U)` is fine with me as well. I can push an update later on to work with a dts extracted info, but all in all, I prefer this than the bindings inferred default `ngpios` value.

Reply:
OK, I'll queue those changes in the next update.

Reply:
Thanks!

Reply:
Done.